src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/SaveTempDataPropertyFilter.cs

```diff
@@ -1,19 +1,32 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using System.Linq;
 using System.Collections.Generic;
 using System.Reflection;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Internal;
 
 namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
 {
-    public class SaveTempDataPropertyFilter : ISaveTempDataCallback
+    public class SaveTempDataPropertyFilter : ISaveTempDataCallback, IActionFilter
     {
-        public string Prefix { get; set; }
+        private readonly ITempDataDictionaryFactory _factory;
+
+        public SaveTempDataPropertyFilter(ITempDataDictionaryFactory factory)
+        {
+            _factory = factory;
+        }
+
+        public string Prefix = "TempDataProperty-";
 
         public object Subject { get; set; }
 
         public IDictionary<PropertyInfo, object> OriginalValues { get; set; }
 
+        internal IList<PropertyHelper> PropertyHelpers { get; set; }
```

Comment:
`PropertyHelper` is an `internal` class. I cannot make the `IList<PropertyHelper>` `public` because then the accessibility is inconsistent. If I set it to `private` I cannot set them in the `SaveTempDataPropertyFilterFactory`. Any suggestions to solve this?